plugins/modules/files/iso_create.py

```diff
@@ -0,0 +1,270 @@
+        # will create intermediate dir for new ISO file
+        try:
+            os.makedirs(dest_iso_dir)
+        except OSError as err:
+            module.fail_json(msg='Exception caught when creating folder %s, with error %s' % (dest_iso_dir, to_text(err)))
+
+    volume_id = module.params.get('vol_ident')
+    inter_level = module.params.get('interchange_level')
+    rock_ridge = module.params.get('rock_ridge')
+    file_mode = module.params.get('file_mode')
+    if rock_ridge is None and file_mode is not None:
+        file_mode = None
+
+    use_joliet = module.params.get('joliet')
+    use_udf = None
+    if 'udf' in module.params and module.params['udf']:
+        use_udf = '2.60'
+
+    result = dict(
+        changed=False,
+        source_file=src_file_list,
+        created_iso=dest_iso,
+        interchange_level=inter_level,
+        vol_ident=volume_id,
+        rock_ridge=rock_ridge,
+        joliet=use_joliet,
+        udf=use_udf
+    )
+    if module.check_mode:
+        module.exit_json(**result)
+
+    iso_file = pycdlib.PyCdlib()
+    iso_file.new(interchange_level=inter_level, vol_ident=volume_id, rock_ridge=rock_ridge, joliet=use_joliet, udf=use_udf)
+
+    for src_file in src_file_list:
+        # if specify a dir then go through the dir to add files and dirs
+        if os.path.isdir(src_file):
+            dir_list = []
+            file_list = []
+            src_file = src_file.rstrip('/')
+            dir_name = os.path.basename(src_file)
+            add_directory(module, iso_file=iso_file, dir_path='/' + dir_name, rock_ridge=rock_ridge,
+                          file_mode=file_mode, use_joliet=use_joliet, use_udf=use_udf)
+
+            # get dir list and file list
+            for path, dirs, files in os.walk(src_file):
+                for filename in files:
+                    file_list.append(os.path.join(path, filename))
+                for dir in dirs:
+                    dir_list.append(os.path.join(path, dir))
+            for new_dir in dir_list:
+                add_directory(module, iso_file=iso_file, dir_path=new_dir.split(os.path.dirname(src_file))[1],
+                              rock_ridge=rock_ridge, file_mode=file_mode, use_joliet=use_joliet, use_udf=use_udf)
+            for new_file in file_list:
+                add_file(module, iso_file=iso_file, src_file=new_file,
+                         file_path=new_file.split(os.path.dirname(src_file))[1], rock_ridge=rock_ridge,
+                         file_mode=file_mode, use_joliet=use_joliet, use_udf=use_udf)
+        # if specify a file then add this file directly to the '/' path in ISO
+        else:
+            add_file(module, iso_file=iso_file, src_file=src_file, file_path='/' + os.path.basename(src_file),
+                     rock_ridge=rock_ridge, file_mode=file_mode, use_joliet=use_joliet, use_udf=use_udf)
+
+    iso_file.write(dest_iso)
+    iso_file.close()
+
+    result['changed'] = True
```

Comment:
Yes, if it comes to this line that the iso file is created without failure. Thanks.

Reply:
what if it already exists? and the module supports check mode

Reply:
Since the existing iso may be created from the files which are not the same as the new configured ones, so now it does not check if the iso file with same name exists, if you configured the same path and iso file name, then the new created iso file will overwrite the existing one.